# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1347385482 14400
#      Tue Sep 11 13:44:42 2012 -0400
# Node ID 9a8dbd6b6b20f31f6c5438393b26414a63f7b0b6
# Parent  a291f850743ca2541caebdbdb93c96b348d34c2b
* mysparse.c (mexFunction): Declare pbr and pbr2 as mxLogical*, not bool*.

diff --git a/examples/mysparse.c b/examples/mysparse.c
--- a/examples/mysparse.c
+++ b/examples/mysparse.c
@@ -51,17 +51,17 @@ mexFunction (int nlhs, mxArray *plhs[], 
       for (i = 0; i < n + 1; i++)
         jc2[i] = jc[i];
 
       if (nlhs > 0)
         plhs[0] = v;
     }
   else if (mxIsLogical (prhs[0]))
     {
-      bool *pbr, *pbr2;
+      mxLogical *pbr, *pbr2;
       mexPrintf ("Matrix is %d-by-%d logical",
                  " sparse matrix", m, n);
       mexPrintf (" with %d elements\n", nz);
 
       pbr = mxGetLogicals (prhs[0]);
       ir = mxGetIr (prhs[0]);
       jc = mxGetJc (prhs[0]);
 
diff --git a/libinterp/interp-core/mexproto.h b/libinterp/interp-core/mexproto.h
--- a/libinterp/interp-core/mexproto.h
+++ b/libinterp/interp-core/mexproto.h
@@ -82,17 +82,18 @@ extern OCTINTERP_API void mexErrMsgTxt (
 extern OCTINTERP_API void mexErrMsgIdAndTxt (const char *id, const char *s, ...);
 extern OCTINTERP_API void mexWarnMsgTxt (const char *s);
 extern OCTINTERP_API void mexWarnMsgIdAndTxt (const char *id, const char *s, ...);
 extern OCTINTERP_API int mexPrintf (const char *fmt, ...);
 
 extern OCTINTERP_API mxArray *mexGetVariable (const char *space, const char *name);
 extern OCTINTERP_API const mxArray *mexGetVariablePtr (const char *space, const char *name);
 
-extern OCTINTERP_API int mexPutVariable (const char *space, const char *name, mxArray *ptr);
+extern OCTINTERP_API int mexPutVariable (const char *space, const char *name,
+                                         const mxArray *ptr);
 
 extern OCTINTERP_API void mexMakeArrayPersistent (mxArray *ptr);
 extern OCTINTERP_API void mexMakeMemoryPersistent (void *ptr);
 
 extern OCTINTERP_API int mexAtExit (void (*f) (void));
 extern OCTINTERP_API const mxArray *mexGet (double handle, const char *property);
 extern OCTINTERP_API int mexIsGlobal (const mxArray *ptr);
 extern OCTINTERP_API int mexIsLocked (void);
